backend/src/api/generate-error.ts

```diff
@@ -0,0 +1,53 @@
+import ApiErrorType from '../enums/api-error-type';
+import ExtendedError from '../interfaces/extended-errors';
+
+/**
+ * Give a message prefix according to an [[ApiErrorType]].
+ *
+ * @param {ApiErrorType} type - The type of error.
+ *
+ * @return {string} - Error message prefix.
+ */
+const getTypeMessagePrefix: Function = (type: ApiErrorType): string => {
+  switch (type) {
+    case ApiErrorType.AuthError:
+      return 'Auth';
+    case ApiErrorType.RouteError:
+      return 'Route';
+    case ApiErrorType.TalkError:
+      return 'Talk';
+    case ApiErrorType.UserError:
+      return 'User';
+    case ApiErrorType.ValidationError:
+      return 'Argument';
+    case ApiErrorType.VoteError:
+      return 'Vote';
+    default:
+      return 'Other';
+  }
+};
+
+/**
+ * Generate an error object from given parameters.
+ * Can be generic and standalone, or to use with `express` server.
+ *
+ * @param {ApiErrorType} type - The type of error.
+ * @param {string} details - For an Error object, detail can be `error.message`.
+ * @param {number} code - The HTTP status code of the error.
+ *
+ * @return {ExtendedError} - `Error` object with parameters applied.
+ */
+const generateError: Function = (
```

Comment:
Type inference may work better here, instead of overriding with Function.

Reply:
Indeed - type inference may work better !

Another solution to implements it in a more Typescript way could be:

```typescript
enum ApiErrorCode { AuthError }

type ApiError = {
    code: ApiErrorCode,
    message: string
}

class ApiErrors {
    public static authError(void): ApiError {
        return {
            code: ApiErrorCode.AuthError,
            message: 'Authentication'
        }
    }
}
```

What are your thoughts? I would love to know how I could improve the handling of those generic error messages. 😄

Reply:
Thanks for the reply, that's enough for our evaluation :) we can talk a bit more when we have the chance.